infrastructure/templates/public-api/main.bicep

```diff
@@ -228,8 +228,18 @@ module postgreSqlServerModule 'components/postgresqlDatabase.bicep' = if (update
 
 var psqlManagedIdentityConnectionStringTemplate = 'Server=${psqlServerFullName}.postgres.database.azure.com;Database=[database_name];Port=5432;User Id=[managed_identity_name]'
 
-resource apiContainerAppManagedIdentity 'Microsoft.ManagedIdentity/userAssignedIdentities@2023-01-31' existing = if (deployContainerApp) {
+resource apiContainerAppManagedIdentity 'Microsoft.ManagedIdentity/userAssignedIdentities@2023-01-31' = {
   name: apiContainerAppManagedIdentityName
+  location: location
+}
+
+module acrPullRoleAssignmentModule 'components/containerRegistryRoleAssignment.bicep' = {
+  name: '${apiContainerAppManagedIdentityName}AcrPullRoleAssignmentDeploy'
+  params: {
+    role: 'AcrPull'
+    containerRegistryName: acrName
+    managedIdentityName: apiContainerAppManagedIdentity.name
+  }
```

Comment:
This is the Api container's registry access so maybe it should include 'apiContainer' in the name in case another service/identity needs a similar role assignment? Something like `apiContainerAcrPullRoleAssignment` or `apiContainerRegistryAccess` rather than `acrPullRoleAssignmentModule`?

Could also think about grouping everything related to the Api container into a new `app/apiContainer.bicep` in future.

Reply:
Updated to `apiContainerAppAcrPullRoleAssignmentModule`.

Also, I already have a ticket for better modularising the Bicep as a whole - https://dfedigital.atlassian.net/browse/EES-5479.  I think this should be done prior to deploying to Test.